src/accel/detail/HitProcessor.cc

```diff
@@ -294,7 +294,7 @@ bool HitProcessor::update_touchable(Real3 const& pos,
     // Check that physical and logical volumes are consistent
     G4VPhysicalVolume* pv = touchable->GetVolume(0);
     CELER_ASSERT(pv);
-    if (!CELER_UNLIKELY(pv->GetLogicalVolume() != lv))
+    if (!CELER_UNLIKELY((pv->GetLogicalVolume() != lv)))
```

Comment:
If this isn't working as intended, maybe the macro itself needs to be fixed?

Reply:
Something like this instead?
```diff
diff --git a/src/corecel/Macros.hh b/src/corecel/Macros.hh
index fc6dd8f6c..6d9faab25 100644
--- a/src/corecel/Macros.hh
+++ b/src/corecel/Macros.hh
@@ -71,7 +71,7 @@
 #    define CELER_UNLIKELY(COND) __builtin_expect(!!(COND), 0)
 #else
 // No other compilers seem to have a similar builtin
-#    define CELER_UNLIKELY(COND) COND
+#    define CELER_UNLIKELY(COND) (COND)
 #endif
 
 /*!
```

Reply:
Yeah. Were you getting a compiler warning?

Reply:
Yep `HitProcessor.cc:297:48: error: logical not is only applied to the left hand side of comparison [-Werror=logical-not-parentheses]`

Reply:
Oooh yes we really need that parens. Thanks warning!

Reply:
(And thanks Amanda 😉 )